<a href="https://colab.research.google.com/github/ty202503149-aming/MIT_django_analytics_project/blob/main/gaming_academic_performances_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [16]:
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [17]:
import pandas as pd

In [18]:
df.head()

,student_id,age,gender,gaming_hours,study_hours,sleep_hours,attendance,gaming_genre,social_activity,device_usage,reaction_time_ms,addiction_score,stress_level,grades
0,1,22,Male,7.23,8.78,6.96,91.44,FPS,3.25,9.36,235.84,14.69,Low,86.459555
1,2,19,Male,0.07,8.72,7.63,63.63,Casual,1.02,3.21,328.71,2.47,Medium,98.230000
2,3,23,Female,1.73,9.56,4.40,83.26,Casual,3.46,5.56,313.61,4.73,High,90.560000
3,4,20,Female,6.62,1.68,7.83,75.04,RPG,1.46,11.78,241.84,14.54,Low,32.670000
4,5,22,Female,5.36,5.83,5.55,65.57,FPS,1.01,8.23,249.31,12.48,Low,58.710000


In [19]:
path = "/content/drive/MyDrive/AL_RAJAB_AMING_TESTING_FINAL_EXAM/Gaming_Academic_Performance.csv"

df = pd.read_csv(path)

df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   student_id        8000 non-null   int64  
 1   age               8000 non-null   int64  
 2   gender            8000 non-null   object 
 3   gaming_hours      8000 non-null   float64
 4   study_hours       8000 non-null   float64
 5   sleep_hours       8000 non-null   float64
 6   attendance        8000 non-null   float64
 7   gaming_genre      8000 non-null   object 
 8   social_activity   8000 non-null   float64
 9   device_usage      8000 non-null   float64
 10  reaction_time_ms  8000 non-null   float64
 11  addiction_score   8000 non-null   float64
 12  stress_level      8000 non-null   object 
 13  grades            8000 non-null   float64
dtypes: float64(9), int64(2), object(3)
memory usage: 875.1+ KB


In [20]:
df_model = df.drop(columns=['student_id'])

In [21]:
categorical_cols = ['gender', 'gaming_genre']

In [22]:
df_encoded = pd.get_dummies(df_model, columns=categorical_cols, drop_first=True)

In [23]:
df_encoded.head()
df_encoded.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   age               8000 non-null   int64  
 1   gaming_hours      8000 non-null   float64
 2   study_hours       8000 non-null   float64
 3   sleep_hours       8000 non-null   float64
 4   attendance        8000 non-null   float64
 5   social_activity   8000 non-null   float64
 6   device_usage      8000 non-null   float64
 7   reaction_time_ms  8000 non-null   float64
 8   addiction_score   8000 non-null   float64
 9   stress_level      8000 non-null   object 
 10  grades            8000 non-null   float64
 11  gender_Male       8000 non-null   bool   
 12  gender_Other      8000 non-null   bool   
 13  gaming_genre_FPS  8000 non-null   bool   
 14  gaming_genre_RPG  8000 non-null   bool   
dtypes: bool(4), float64(9), int64(1), object(1)
memory usage: 718.9+ KB


In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [ ]:
y_addiction = df_encoded['addiction_score']

In [ ]:
X_addiction = df_encoded.drop(columns=['addiction_score', 'grades'])

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction,
    test_size=0.2,
    random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

mse_a = mean_squared_error(y_test_a, y_pred_a)
r2_a = r2_score(y_test_a, y_pred_a)

print("Addiction regression MSE:", mse_a)
print("Addiction regression R^2:", r2_a)

ValueError: could not convert string to float: 'Medium'

In [ ]:
X_addiction = df_encoded.drop(columns=['addiction_score', 'grades'])

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction,
    test_size=0.2,
    random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

mse_a = mean_squared_error(y_test_a, y_pred_a)
r2_a = r2_score(y_test_a, y_pred_a)

print("Addiction regression MSE:", mse_a)
print("Addiction regression R^2:", r2_a)

ValueError: could not convert string to float: 'Medium'

In [ ]:
y_addiction = pd.to_numeric(df_encoded['addiction_score'], errors='coerce')

In [ ]:
data = df_encoded.copy()
data['addiction_score'] = pd.to_numeric(data['addiction_score'], errors='coerce')
data = data.dropna(subset=['addiction_score'])

y_addiction = data['addiction_score']
X_addiction = data.drop(columns=['addiction_score', 'grades'])

In [ ]:
stress_map = {'Low': 0, 'Medium': 1, 'High': 2}
df_encoded['stress_level_num'] = df_encoded['stress_level'].map(stress_map)

In [ ]:
data = df_encoded.copy()
data['addiction_score'] = pd.to_numeric(data['addiction_score'], errors='coerce')
data = data.dropna(subset=['addiction_score'])

y_addiction = data['addiction_score']
X_addiction = data.drop(columns=['addiction_score', 'grades', 'stress_level'], errors='ignore')

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction, test_size=0.2, random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

mse_a = mean_squared_error(y_test_a, y_pred_a)
r2_a = r2_score(y_test_a, y_pred_a)

print("Addiction regression MSE:", mse_a)
print("Addiction regression R^2:", r2_a)

Addiction regression MSE: 3.8377395969771557
Addiction regression R^2: 0.848651416506023


In [ ]:
data = df_encoded.copy()

In [ ]:
data['addiction_score'] = pd.to_numeric(data['addiction_score'], errors='coerce')

In [ ]:
data = data.dropna(subset=['addiction_score'])

In [ ]:
y_addiction = data['addiction_score']
X_addiction = data.drop(columns=['addiction_score', 'grades', 'stress_level'], errors='ignore')

In [ ]:
X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction,
    test_size=0.2,
    random_state=42
)

In [ ]:
reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

LinearRegression()

In [ ]:
y_pred_a = reg_addiction.predict(X_test_a)

In [ ]:
mse_a = mean_squared_error(y_test_a, y_pred_a)
r2_a = r2_score(y_test_a, y_pred_a)

print("Addiction regression MSE:", mse_a)
print("Addiction regression R^2:", r2_a)

Addiction regression MSE: 3.8377395969771557
Addiction regression R^2: 0.848651416506023


I just done the addiction_score regression cell and check the output.

In [29]:
y_addiction = df_encoded['addiction_score']

In [27]:
path = "/content/drive/MyDrive/AL_RAJAB_AMING_TESTING_FINAL_EXAM/Gaming_Academic_Performance.csv"
import pandas as pd
df = pd.read_csv(path)
df.head()

,student_id,age,gender,gaming_hours,study_hours,sleep_hours,attendance,gaming_genre,social_activity,device_usage,reaction_time_ms,addiction_score,stress_level,grades
0,1,22,Male,7.23,8.78,6.96,91.44,FPS,3.25,9.36,235.84,14.69,Low,86.459555
1,2,19,Male,0.07,8.72,7.63,63.63,Casual,1.02,3.21,328.71,2.47,Medium,98.230000
2,3,23,Female,1.73,9.56,4.40,83.26,Casual,3.46,5.56,313.61,4.73,High,90.560000
3,4,20,Female,6.62,1.68,7.83,75.04,RPG,1.46,11.78,241.84,14.54,Low,32.670000
4,5,22,Female,5.36,5.83,5.55,65.57,FPS,1.01,8.23,249.31,12.48,Low,58.710000


In [31]:
from google.colab import sheets
sheet = sheets.InteractiveSheet(df=df)

https://docs.google.com/spreadsheets/d/1w9s-s1nwzWZBm8RquU8ugm8VYrmCl6TH4Odu2biYA0E/edit#gid=0


In [32]:
X_addiction = df_encoded.drop(columns=['addiction_score', 'grades', 'stress_level'], errors='ignore')

In [33]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction, test_size=0.2, random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

print("MSE:", mean_squared_error(y_test_a, y_pred_a))
print("R^2:", r2_score(y_test_a, y_pred_a))

MSE: 3.8366517338952235
R^2: 0.8486943184623243


In [34]:
y_addiction = df_encoded['addiction_score']

In [35]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction, test_size=0.2, random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

print("MSE:", mean_squared_error(y_test_a, y_pred_a))
print("R^2:", r2_score(y_test_a, y_pred_a))

MSE: 3.8366517338952235
R^2: 0.8486943184623243


In [36]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction, test_size=0.2, random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

print("MSE:", mean_squared_error(y_test_a, y_pred_a))
print("R^2:", r2_score(y_test_a, y_pred_a))

MSE: 3.8366517338952235
R^2: 0.8486943184623243


In [37]:
y_addiction = df_encoded['addiction_score']
X_addiction = df_encoded.drop(columns=['addiction_score', 'grades', 'stress_level'], errors='ignore')

In [38]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

X_train_a, X_test_a, y_train_a, y_test_a = train_test_split(
    X_addiction, y_addiction, test_size=0.2, random_state=42
)

reg_addiction = LinearRegression()
reg_addiction.fit(X_train_a, y_train_a)

y_pred_a = reg_addiction.predict(X_test_a)

print("MSE:", mean_squared_error(y_test_a, y_pred_a))
print("R^2:", r2_score(y_test_a, y_pred_a))

MSE: 3.8366517338952235
R^2: 0.8486943184623243


In [39]:
y_grades = df_encoded['grades']
X_grades = df_encoded.drop(columns=['addiction_score', 'grades', 'stress_level'], errors='ignore')

In [40]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

X_train_g, X_test_g, y_train_g, y_test_g = train_test_split(
    X_grades, y_grades, test_size=0.2, random_state=42
)

reg_grades = LinearRegression()
reg_grades.fit(X_train_g, y_train_g)

y_pred_g = reg_grades.predict(X_test_g)

print("Grades MSE:", mean_squared_error(y_test_g, y_pred_g))
print("Grades R^2:", r2_score(y_test_g, y_pred_g))

Grades MSE: 50.54691264058169
Grades R^2: 0.8991583945626934


Output of reggression: grades.

NEXT: Classification of Stress_level

In [42]:
stress_map = {'Low': 0, 'Medium': 1, 'High': 2}
df_encoded['stress_level_num'] = df_encoded['stress_level'].map(stress_map)

In [43]:
y_stress = df_encoded['stress_level_num']
X_stress = df_encoded.drop(columns=['addiction_score', 'grades', 'stress_level', 'stress_level_num'], errors='ignore')

In [44]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report

X_train_s, X_test_s, y_train_s, y_test_s = train_test_split(
    X_stress, y_stress, test_size=0.2, random_state=42, stratify=y_stress
)

clf_stress = LogisticRegression(max_iter=1000)
clf_stress.fit(X_train_s, y_train_s)

y_pred_s = clf_stress.predict(X_test_s)

print("Accuracy:", accuracy_score(y_test_s, y_pred_s))
print(classification_report(y_test_s, y_pred_s))

Accuracy: 0.92625
              precision    recall  f1-score   support

           0       0.95      0.94      0.94       549
           1       0.96      0.96      0.96       849
           2       0.75      0.75      0.75       202

    accuracy                           0.93      1600
   macro avg       0.88      0.88      0.88      1600
weighted avg       0.93      0.93      0.93      1600



/usr/local/lib/python3.12/dist-packages/sklearn/linear_model/_logistic.py:465: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


We will try the single cell only.

In [45]:
stress_map = {'Low': 0, 'Medium': 1, 'High': 2}
df_encoded['stress_level_num'] = df_encoded['stress_level'].map(stress_map)

In [46]:
y_stress = df_encoded['stress_level_num']
X_stress = df_encoded.drop(columns=['addiction_score', 'grades', 'stress_level', 'stress_level_num'], errors='ignore')

Next we will train:

In [47]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report

X_train_s, X_test_s, y_train_s, y_test_s = train_test_split(
    X_stress, y_stress, test_size=0.2, random_state=42, stratify=y_stress
)

clf_stress = LogisticRegression(max_iter=1000)
clf_stress.fit(X_train_s, y_train_s)

y_pred_s = clf_stress.predict(X_test_s)

print("Accuracy:", accuracy_score(y_test_s, y_pred_s))
print(classification_report(y_test_s, y_pred_s))

Accuracy: 0.92625
              precision    recall  f1-score   support

           0       0.95      0.94      0.94       549
           1       0.96      0.96      0.96       849
           2       0.75      0.75      0.75       202

    accuracy                           0.93      1600
   macro avg       0.88      0.88      0.88      1600
weighted avg       0.93      0.93      0.93      1600



/usr/local/lib/python3.12/dist-packages/sklearn/linear_model/_logistic.py:465: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [48]:
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

clf_stress = make_pipeline(
    StandardScaler(),
    LogisticRegression(max_iter=3000)
)

clf_stress.fit(X_train_s, y_train_s)
y_pred_s = clf_stress.predict(X_test_s)

print("Accuracy:", accuracy_score(y_test_s, y_pred_s))
print(classification_report(y_test_s, y_pred_s))

Accuracy: 0.933125
              precision    recall  f1-score   support

           0       0.96      0.94      0.95       549
           1       0.96      0.97      0.96       849
           2       0.75      0.76      0.75       202

    accuracy                           0.93      1600
   macro avg       0.89      0.89      0.89      1600
weighted avg       0.93      0.93      0.93      1600



strong classification result overall. The model gets about 93.3% accuracy, which means it predicts the stress level correctly most of the time.